apps/internal/oauth/ops/accesstokens/accesstokens.go

```diff
@@ -140,10 +155,17 @@ func (c *Credential) JWT(ctx context.Context, authParams authority.AuthParams) (
 
 // thumbprint runs the asn1.Der bytes through sha1 for use in the x5t parameter of JWT.
 // https://tools.ietf.org/html/rfc7517#section-4.8
-func thumbprint(cert *x509.Certificate) []byte {
-	/* #nosec */
-	a := sha1.Sum(cert.Raw)
-	return a[:]
+func thumbprint(cert *x509.Certificate, alg string) []byte {
+	switch alg {
+	case jwt.SigningMethodRS256.Name:
+		/* #nosec */
```

Comment:
Please add a comment that some identity providers like ADFS don't support SHA256 assertions, as a justification for the static analysis warnings.

Reply:
Added a comment